src/iden3comm/types/protocol/discovery-protocol.ts

```diff
@@ -0,0 +1,37 @@
+import { PROTOCOL_MESSAGE_TYPE } from '../../constants';
+import { BasicMessage } from '../packer';
+
+/** @beta DiscoveryProtocolFeatureType is enum for supported feature-types */
+export enum DiscoveryProtocolFeatureType {
+  Accept = 'accept'
+}
+
+/** @beta DiscoverFeatureQueriesMessage is struct the represents discover feature queries message */
+export type DiscoverFeatureQueriesMessage = BasicMessage & {
+  body: DiscoverFeatureQueriesBody;
+  type: typeof PROTOCOL_MESSAGE_TYPE.DISCOVERY_PROTOCOL_QUERIES_MESSAGE_TYPE;
+};
+
+/** @beta DiscoverFeatureQueriesBody is struct the represents discover feature queries body */
+export type DiscoverFeatureQueriesBody = {
+  queries: {
+    'feature-type': DiscoveryProtocolFeatureType | string;
```

Comment:
Maybe better to declare  enum for that something like `DiscoverFeatureQueryType`? and use across all 'feature-type' occurences

Reply:
Sould we allow to pass `DiscoveryProtocolFeatureType | string` arbitrary string here?